crates/ra_ide_api/src/goto_definition.rs

```diff
@@ -122,9 +123,40 @@ pub(crate) fn reference_definition(
             Some(Resolution::SelfType(_impl_block)) => {
                 // TODO: go to the implemented type
             }
-            None => {}
+            None => {
+                // If we failed to resolve then check associated items
+                if let Some(function) = function {
+                    // Should we do this above and then grab path from the PathExpr?
+                    if let Some(path_expr) =
+                        name_ref.syntax().ancestors().find_map(ast::PathExpr::cast)
+                    {
+                        let infer_result = function.infer(db);
+                        let source_map = function.body_source_map(db);
+                        let expr = ast::Expr::cast(path_expr.syntax()).unwrap();
+
+                        if let Some(res) = source_map
+                            .node_expr(expr)
+                            .and_then(|it| infer_result.assoc_resolutions(it.into()))
+                        {
+                            match res {
```

Comment:
There should probably be a NavigationTarget::from_impl_item?

Reply:
Done